### PR TITLE
Cw 09 implement picker screen with ability to search

### PR DESCRIPTION
**Pull request by Serhii-Borodenko**
-Created a new selection screen
-Added a search bar with the ability to search by name and full name of currencies
-Added cryptocurrency icons
-Added Cancel button to cancel search action
- Updated all localization files

### CODE REVIEW

**Comment by mkyq** [comment]
@Serhii-Borodenko please update your PR with last changes from `main` and resolve merge conflicts

**Review by mkyq — CHANGES_REQUESTED** [review]

File: lib/src/screens/exchange/widgets/currency_picker.dart [hunk 1]
```diff
@@ -43,6 +43,15 @@ class CurrencyPickerState extends State<CurrencyPicker> {
...
   TextStyle appBarTextStyle;
+  var pickerItemsList = CryptoCurrency.all
+      .map((CryptoCurrency cur) => PickerItem<CryptoCurrency>(
+            cur,
+            title: PickerItem.titleForCurrency(cur),
+            iconPath: PickerItem.iconPathForCurrency(cur),
+            tag: PickerItem.tagForCurrency(cur),
+            description: PickerItem.descriptionForCurrency(cur)
+          ))
+      .toList();
```
Comment (mkyq) [inline lib/src/screens/exchange/widgets/currency_picker.dart:54]: Put that in initState method

**Review by mkyq — CHANGES_REQUESTED** [review]

**Comment by mkyq** [comment]
cc @Serhii-Borodenko

File: lib/src/screens/exchange/widgets/currency_picker.dart [hunk 2]
```diff
@@ -9,65 +10,72 @@ import 'package:cake_wallet/src/widgets/alert_background.dart';
...
-            description: PickerItem.descriptionForCurrency(cur)
-          ))
-      .toList();
 
-  void currencySearchBySubstring(String subString, List<CryptoCurrency> list) {
-    subCryptoCurrencyList = [];
+  List<PickerItem<CryptoCurrency>> get cleanSubPickerItemsList =>
+      subPickerItemsList = pickerItemsList
+          .where((element) => items.contains(element.original))
+          .toList();
```
Comment (mkyq) [inline lib/src/screens/exchange/widgets/currency_picker.dart:66]: Why is that getter ? It should be just a function.

File: lib/src/screens/exchange/widgets/currency_picker.dart [hunk 3]
```diff
@@ -147,15 +156,30 @@ class CurrencyPickerState extends State<CurrencyPicker> {
...
+                        setState(() {
+                          widget.selectedAtIndex = index;
+                        });
+                        widget
+                            .onItemSelected(subPickerItemsList[index].original);
+
+                        print('isConvertFrom = ${widget.isConvertFrom}');
+                        print('isMoneroWallet = ${!widget.isMoneroWallet}');
+                        print(
+                            'ryptoCurrency.xmr = ${subPickerItemsList[index].original == CryptoCurrency.xmr}');
```
Comment (mkyq) [inline lib/src/screens/exchange/widgets/currency_picker.dart:173]: Remove prints

**Review by mkyq — CHANGES_REQUESTED** [review]

**Comment by mkyq** [comment]
cc @Serhii-Borodenko

**Comment by Serhii-Borodenko** [comment]
done

**Pull request merged**